frame/dex/src/lib.rs

```diff
@@ -424,41 +435,27 @@ pub mod pallet {
 		) -> DispatchResult {
 			let sender = ensure_signed(origin)?;
 
-			let pool_id = Self::get_pool_id(asset1, asset2);
-
 			ensure!(
 				amount_in > Zero::zero() && amount_out_min > Zero::zero(),
 				Error::<T>::ZeroAmount
 			);
+			ensure!(path.len() >= 2, Error::<T>::InvalidPath);
 
 			let now = frame_system::Pallet::<T>::block_number();
 			ensure!(deadline >= now, Error::<T>::DeadlinePassed);
 
-			let pool_account = Self::get_pool_account(pool_id);
-			let balance1 = Self::get_balance(&pool_account, asset1);
-			let balance2 = Self::get_balance(&pool_account, asset2);
-			if balance1.is_zero() {
-				return Err(Error::<T>::PoolNotFound.into())
-			}
-
-			let amount_out = Self::get_amount_out(&amount_in, &balance1, &balance2)?;
-
-			ensure!(amount_out >= amount_out_min, Error::<T>::InsufficientOutputAmount);
+			let amounts = Self::get_amounts_out(&amount_in, &path)?;
+			let amount_out = amounts.last().expect("Has always more than 1 element");
```

Comment:
Deref it here will work nicer:
```suggestion
			let amount_out = *amounts.last().expect("Has always more than 1 element");
```